pandas/io/excel.py

```diff
@@ -72,7 +72,7 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
-def read_excel(io, sheetname=0, header=0, skiprows=None, skip_footer=0,
+def read_excel(io, sheet_name=0, header=0, skiprows=None, skip_footer=0,
```

Comment:
so this will actually need to be deprecated, so re-add ``sheetname=None``. Then if ``sheetname`` is not None, you will issue a warning that it is deprecated and reset ``sheet_name=sheetname``.

This way if people pass as positional args it will work, and if they pass as a kwarg then it will show the deprecation warning (but still work).

Pls add a tests for this as well.